hs_core/templates/pages/genericresource.html

```diff
@@ -828,13 +831,34 @@ <h3>Resource Specific</h3>
       const SUPPORTE_FILE_TYPES = "{{ cm.get_supported_upload_file_types }}";
       const ALLLOW_MULTIPLE_FILE_UPLOAD = "{{ cm.allow_multiple_file_upload }}";
       const RESOURCE_MODE = "{% if resource_edit_mode %}Edit{% else %}View{% endif %}";
-      const CURRENT_USER_ID = {{ request.user.id }};
+      const CURRENT_USER_ID = {{ request.user.id|default:"null" }};
```

Comment:
I really think these blocks should be set from an ajax call rather than the intial page template.  It should also speed up our initial response quite a bit.

Reply:
We could try this out once all the logic has been moved to Vue, so we know exactly what data and what endpoints we need.

However, as I understand this, the AJAX calls will be made after the client receives the HTML. Without the data the page is unusable so this will actually slow down loading the page until the AJAX calls are resolved. 

We'll have to do some profiling to see which method is better.

Reply:
the page would not be complete, but the browser should be able to start to paint the page while concurrently with the ajax call to get this data.  Unless Vue behaves differently than I think it does.